src/scanpy/tools/_umap.py

```diff
@@ -267,14 +333,154 @@ def umap(
             random_state=random_state,
         )
         X_umap = umap.fit_transform(X_contiguous)
+
     adata.obsm[key_obsm] = X_umap  # annotate samples with UMAP coordinates
     logg.info(
         "    finished",
         time=start,
         deep=(
             "added\n"
-            f"    {key_obsm!r}, UMAP coordinates (adata.obsm)\n"
-            f"    {key_uns!r}, UMAP parameters (adata.uns)"
+            f"    {key_obsm!r}, {method_name} coordinates (adata.obsm)\n"
+            f"    {key_uns!r}, {method_name} parameters (adata.uns)"
         ),
     )
     return adata if copy else None
+
+
+# Convenience function for densMAP
+
+
+def densmap(
```

Comment:
I’d rather not maintain that. Just `method='densmap'` should be sufficient, but if `densmap = partial(umap, method='densmap')` works and displays in the docs as expected, I’d consider it.

If you want to make it more visible, you could add an example to the `sc.pl.umap` docs and maybe even a tutorial about what it’s for and how to best use it

Reply:
I tried the `partial` option, but there seem to be issues with inference of the types from the parent `umap` function when generating the autodoc for sphinx that are not straightforward to resolve.